src/main/java/com/aws/greengrass/clientdevices/auth/PermissionEvaluationUtils.java

```diff
@@ -134,9 +138,27 @@ private boolean compareResource(Resource requestResource, String policyResource)
             return true;
         }
 
-        WildcardTrie wildcardTrie = new WildcardTrie();
-        wildcardTrie.add(policyResource);
-        return wildcardTrie.matchesStandard(requestResource.getResourceStr());
+        if (matchMqttWildcards()) {
```

Comment:
Switching comment to threads to make it more sane.


My point is that IoT Core for MQTT policies does not let you subscribe to A/B when the policy is `A/#`. Greengrass PubSub does allow this, but CDA does not in order to match the behavior of IoT Core.

Reply:
Okay yeah I understand your point now.  So to have parity with IoT Core then, `?` single char matching would be needed, so then CDA knows about `*` and `?`

Reply:
Yeah guess so, if iot core just added that in.

Reply:
But wildcards `#` and `+` shouldn't change the way they currently work.

Reply:
Yep will either close or revise this. Reason it came up was customer ask so was exploring the idea